integration-cli: can no longer run integration-cli tests using TLS against a remote docker host

In docker 1.8 I could run the integration-cli tests against a *remote* host - but this no longer works since docker 1.9, e.g.:
```
export DOCKER_HOST=tcp://remoteserver:2376
export DOCKER_CERT_PATH=/home/user/.docker/machine/machines/default
export DOCKER_TLS_VERIFY=1
$ go test -v -check.f TestInfo
panic: Init failed to get version: unexpected EOF. Res=<nil>

goroutine 1 [running]:
github.com/docker/docker/integration-cli.init.2()
	/Users/toddw/goroot/src/github.com/docker/docker/integration-cli/docker_utils.go:58 +0x467
github.com/docker/docker/integration-cli.init()
	/Users/toddw/goroot/src/github.com/docker/docker/integration-cli/docker_hub_pull_suite_test.go:90 +0x871
main.init()
	github.com/docker/docker/integration-cli/_test/_testmain.go:56 +0x4a

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 19 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37
exit status 2
FAIL	github.com/docker/docker/integration-cli	242.110s
```

The test above is getting an EOF error because the server in question is expecting a TLS handshake, but instead it's getting a straight HTTP request for the docker test code.

I'll submit a PR that fixes this problem (or, at least it works for me).

--- a/integration-cli/docker_utils.go
+++ b/integration-cli/docker_utils.go
@@ -3,6 +3,7 @@ package main
 import (
 	"bufio"
 	"bytes"
+	"crypto/tls"
 	"encoding/json"
 	"errors"
 	"fmt"
@@ -481,6 +482,26 @@ func daemonHost() string {
 	return daemonURLStr
 }
 
+func getTLSConfig() (*tls.Config, error) {
+	dockerCertPath := os.Getenv("DOCKER_CERT_PATH")
+
+	if dockerCertPath == "" {
+		return nil, fmt.Errorf("DOCKER_TLS_VERIFY specified, but no DOCKER_CERT_PATH environment variable")
+	}
+
+	option := &tlsconfig.Options{
+		CAFile:   filepath.Join(dockerCertPath, "ca.pem"),
+		CertFile: filepath.Join(dockerCertPath, "cert.pem"),
+		KeyFile:  filepath.Join(dockerCertPath, "key.pem"),
+	}
+	tlsConfig, err := tlsconfig.Client(*option)
+	if err != nil {
+		return nil, err
+	}
+
+	return tlsConfig, nil
+}
+
 func sockConn(timeout time.Duration) (net.Conn, error) {
 	daemon := daemonHost()
 	daemonURL, err := url.Parse(daemon)
@@ -493,6 +514,15 @@ func sockConn(timeout time.Duration) (net.Conn, error) {
 	case "unix":
 		return net.DialTimeout(daemonURL.Scheme, daemonURL.Path, timeout)
 	case "tcp":
+		if os.Getenv("DOCKER_TLS_VERIFY") != "" {
+			// Setup the socket TLS configuration.
+			tlsConfig, err := getTLSConfig()
+			if err != nil {
+				return nil, err
+			}
+			dialer := &net.Dialer{Timeout: timeout}
+			return tls.DialWithDialer(dialer, daemonURL.Scheme, daemonURL.Host, tlsConfig)
+		}
 		return net.DialTimeout(daemonURL.Scheme, daemonURL.Host, timeout)
 	default:
 		return c, fmt.Errorf("unknown scheme %v (%s)", daemonURL.Scheme, daemon)

